kernels/streamer_alu/streamer_add_stream.preprocfinal.mlir

```diff
@@ -19,18 +19,14 @@ func.func public @streamer_add(%arg0 : memref<?xi64, "L3">, %arg1 : memref<?xi64
   "snax.cluster_sync_op"() : () -> ()
   "scf.if"(%0) ({
 
-    %x = "memref.extract_aligned_pointer_as_index" (%4) : (memref<?xi64, "L1">) -> (index)
-    %y = "memref.extract_aligned_pointer_as_index" (%7) : (memref<?xi64, "L1">) -> (index)
-    %z = "memref.extract_aligned_pointer_as_index" (%10) : (memref<?xi64, "L1">) -> (index)
-
-    "snax_stream.streaming_region"(%x, %y, %z) <{
-          "stride_pattern" = [
-                  #snax_stream.stride_pattern<ub = [4], ts = [32], ss = [8]>, 
-                  #snax_stream.stride_pattern<ub = [4], ts = [32], ss = [8]>, 
-                  #snax_stream.stride_pattern<ub = [4], ts = [32], ss = [8]>
-          ], "operandSegmentSizes" = array<i32: 2, 1>,
-          "accelerator" = "snax_alu"}> ({
-    ^0(%103 : !stream.readable<i64>, %104 : !stream.readable<i64>, %105 : !stream.writable<i64>):
+    memref_stream.streaming_region {
+      patterns = [
+          #memref_stream.stride_pattern<ub = [4, 4], index_map = (d0, d1) -> (4 * d0 + d1)>,
+          #memref_stream.stride_pattern<ub = [4, 4], index_map = (d0, d1) -> (4 * d0 + d1)>,
+          #memref_stream.stride_pattern<ub = [4, 4], index_map = (d0, d1) -> (4 * d0 + d1)>
+      ]
+    } ins(%4, %7 : memref<?xi64, "L1">, memref<?xi64, "L1">) outs(%10 : memref<?xi64, "L1">) attrs = {accelerator="snax_alu"} {
+    ^bb0(%103: !stream.readable<i64>, %104: !stream.readable<i64>, %105: !stream.writable<i64>):
```

Comment:
unrelated (?) do we need a memory space attribute in the stream type?

Reply:
I don't think so, as a stream is associated with an accelerator. Maybe just a check when a memref is converted to a stream, that it is in the correct memory space for that accelerator.

Reply:
But if we are planning to use this dialect also for DMA regions we should add support for it, no?